core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1655,6 +1663,217 @@ class KafkaApis(val requestChannel: RequestChannel,
+          new DescribeAclsResponse(throttleTimeMs, null, returnedAcls)
+        sendResponseMaybeThrottle(request, createResponse)
+    }
+  }
+
+  /**
+    * Convert an ACL binding filter to a Scala object.
+    * All ACL and resource fields must be specified (no UNKNOWN, ANY, or null fields are allowed.)
+    *
+    * @param filter     The binding filter as a Java object.
+    * @return           The binding filter as a scala object, or an exception if there was an error
+    *                   converting the Java object.
+    */
+  def toScala(filter: AclBindingFilter) : Try[(Resource, Acl)] = {
+    filter.resourceFilter().resourceType() match {
+      case AdminResourceType.UNKNOWN => return Failure(new InvalidRequestException("Invalid UNKNOWN resource type"))
+      case AdminResourceType.ANY => return Failure(new InvalidRequestException("Invalid ANY resource type"))
+      case _ => {}
+    }
+    var resourceType: ResourceType = null
+    try {
+      resourceType = ResourceType.fromString(filter.resourceFilter().resourceType().toString)
+    } catch {
+      case throwable: Throwable => return Failure(new InvalidRequestException("Invalid resource type"))
+    }
+    var principal: KafkaPrincipal = null
+    try {
+      principal = KafkaPrincipal.fromString(filter.entryFilter().principal())
+    } catch {
+      case throwable: Throwable => return Failure(new InvalidRequestException("Invalid principal"))
+    }
+    filter.entryFilter().operation() match {
+      case AclOperation.UNKNOWN => return Failure(new InvalidRequestException("Invalid UNKNOWN operation type"))
+      case AclOperation.ANY => return Failure(new InvalidRequestException("Invalid ANY operation type"))
+      case _ => {}
+    }
+    val operation = Operation.fromJava(filter.entryFilter().operation()) match {
+      case Failure(throwable) => return Failure(new InvalidRequestException(throwable.getMessage))
+      case Success(op) => op
+    }
+    filter.entryFilter().permissionType() match {
+      case AclPermissionType.UNKNOWN => new InvalidRequestException("Invalid UNKNOWN permission type")
+      case AclPermissionType.ANY => new InvalidRequestException("Invalid ANY permission type")
+      case _ => {}
+    }
+    val permissionType = PermissionType.fromJava(filter.entryFilter.permissionType) match {
+      case Failure(throwable) => return Failure(new InvalidRequestException(throwable.getMessage))
+      case Success(perm) => perm
+    }
+    return Success((Resource(resourceType, filter.resourceFilter().name()), Acl(principal, permissionType,
+                   filter.entryFilter().host(), operation)))
+  }
+
+  /**
+    * Convert a Scala ACL binding to a Java object.
+    *
+    * @param acl        The binding as a Scala object.
+    * @return           The binding as a Java object.
+    */
+  def toJava(acl: (Resource, Acl)) : AclBinding = {
+    acl match {
+      case (resource, acl) =>
```

Comment:
If there's a single `case`, there's no reason to use `match`. You can simply do:

```scala
val (resource, acl) = acl
```

Although there's a name clash, so maybe the parameter should be `aclBinding` or something.